dask/typing.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import annotations
+
+from typing import Any, Callable, Mapping, MutableMapping, Protocol, runtime_checkable
+
+try:
+    from IPython.display import DisplayObject
+except ImportError:
+    DisplayObject = object
+
+
+@runtime_checkable
+class DaskCollection(Protocol):
+    def __dask_graph__(self) -> Mapping | None:
+        pass
+
+    def __dask_keys__(self) -> list:
+        pass
+
+    def __dask_postcompute__(self) -> tuple[Callable, tuple]:
+        pass
+
+    def __dask_postpersist__(self) -> tuple[Callable, tuple]:
+        pass
+
+    def __dask_tokenize__(self) -> Any:
```

Comment:
Would it be fair to say that this should return a `str`, or at least a `Hashable`?